File: hrms/src/main/java/com/kaya/hrms/business/abstracts/JobSeekerService.java
package com.kaya.hrms.business.abstracts;

import java.util.List;

import com.kaya.hrms.core.utilities.results.DataResult;
import com.kaya.hrms.core.utilities.results.Result;
import com.kaya.hrms.entities.Dtos.CvDto;
import com.kaya.hrms.entities.Dtos.JobSeekerDtos.JobSeekerUpdateDto;
import com.kaya.hrms.entities.concretes.JobSeeker;

public interface JobSeekerService {
	
	DataResult<List<JobSeeker>> getAll();

	DataResult<JobSeeker> getById(int jobSeekerId);
	
	DataResult<CvDto> getJobSeekerResumeByJobSeekerId(int jobSeekerId);
	
	Result add(JobSeeker jobSeeker);
	
	Result delete(int jobSeekerId);
	
	Result update(int jobSeekerId, JobSeekerUpdateDto jobSeekerUpdateDto);
	
	Result checkUserExists(String email, String nationalityId);
	
	
}

File: hrmsfrontend/src/services/jobSeekerServices/jobSeekerService.js
import axios from "axios";

export default class JobSeekerService {
  constructor() {
    this.apiUrl = "http://localhost:8080/api/jobSeekersController/";
  }

  getall() {
    return axios.get(`${this.apiUrl}getall`);
  }

  getById(jobSeekerId) {
    return axios.get(`${this.apiUrl}getById?jobSeekerId=${jobSeekerId}`);
  }


  getJobSeekerResumeByJobSeekerId(jobSeekerId) {
    return axios.get(`${this.apiUrl}getJobSeekerResumeByJobSeekerId?jobSeekerId=${jobSeekerId}`);
  }

  update(jobSeekerId, jobSeeker) {
    return axios.put(`${this.apiUrl}/update?jobSeekerId=${jobSeekerId}`, jobSeeker);
  }
}
